[DOCS] Update Instances of Source Data to Data Source (#8857)
great-expectations/great_expectations: docs/docusaurus/docs/guides/expectations/advanced/how_to_compare_two_tables_with_the_onboarding_data_assistant.md

In order to profile our first table and validate our second table, we need to set up our Batch Requests pointing to each set of data.

> In this guide, we will use a MySQL <TechnicalTag tag="datasource" text= "Data Source" /> as our Data Source -- the data we trust to be correct.

Reviewer: Should we be using this `TechnicalTag` **every** time we refer to Data Source (and Data Asset or any other key word)?

Author: @r34ctor using a TechnicalTag every time we refer to Data Source or any other keyword will annoy users quickly. A TechnicalTag should be used discriminately. Again, the joys of legacy docs.

> In this guide, we will use a MySQL <TechnicalTag tag="datasource" text= "Data Source" /> as our Data Source -- the data we trust to be correct.

Reviewer: ```suggestion
In this guide, you'll use a MySQL <TechnicalTag tag="datasource" text= "Data Source" />.
```
